ruby/basics/conditional_logic.md

```diff
@@ -1,43 +1,246 @@
+`7 > 5 #=> true`
+
+`<`(less than) returns true if the value on the left of the operator is smaller than the value on the right.
+`5 < 7 #=> true`
+
+`>=`(greater than or equal to) returns true if the value on the left of the operator is larger than or equal to the value on the right.
+`7 > 5 #=> true`
+
+`<=`(less than or equal to) returns true if the value on the left of the operator is smaller than or equal to the value on the right.
+`5 < 7 #=> true`
+
+`.eql?` checks both the value type and actual value it holds.
+`5.eql?(5.0) #=> false` - because although they are the same value, one is an integer and one is a float
 
-## Booelean Logic
-- basics of boolean logic
-- equality operators
-- comparison operators
-- logical operators
-- order of logic
+`.equal?` checks both values are the exact same object in memory. This can be slightly confusing because of the way computers store some values for efficiency. Two variables pointing to a number will usually always return true.
+```ruby
+a = 5
+b = 5
+a.equal?(b) #=> true
+```
+This is true because of the way computers store integers in memory. So although you have set two different variables holding the number 5, they point to the same object in memory. However, consider the next code example.
+```ruby
+a = "hello"
+b = "hello"
+a.equal?(b) #=> false
+```
+This happens because computers can't store strings efficiently in the same way as they store numbers. So although the values of the variables are the same. The computer will have created two separate string objects in memory.
+
+All of the above operators also work on data types other than numbers such as strings. Why not have a play around in a repl?
+
+## Logical Operators
+Sometimes you'll want to write an expression that contains more than one condition. That's accomplished in Ruby with Logical Operators.
+
+The operators are `&&`(and), `||`(or) and `!`(not).
+
+There are some differences between the word versions and their symbolic equivalents, particulary the way they evaluate code. I recommend you read [this piece](http://www.virtuouscode.com/2010/08/02/using-and-and-or-in-ruby/) that will explain the differences.
+
+The `&&` operator returns true if **both** the left and right expression return true.
+```ruby
+if 1 < 2 && 5 < 6
+  puts "Party at Kevins"
+end
+
+# This can also be written as
+if 1 < 2 and 5 < 6
+```
+One thing to keep in mind with the `&&` operator is the order of logic. The expressions are evaluated from left to right and because both expressions must return true, if the first expression encountered returns false, the second expression is never checked because, to the Ruby interpreter, it's pointless as the expression can never return true. This is known as short circuit evaluation.
+
+The `||` operator returns true if either the left or right expression return true. This means both expressions must always be checked, unlike with the and operator above.
```

Comment:
Yeah Ruby uses lazy evaluation with OR statements. If the left side of the OR is true it will return true without checking if the right side of the OR is true or false.